src/Chat/OpenAIChat.php

```diff
@@ -82,17 +87,16 @@ public function getTotalTokens(): int
 
     public function generateTextOrReturnFunctionCalled(string $prompt): string|FunctionInfo
     {
+        $this->functionsCalled = [];
         $this->lastFunctionCalled = null;
-        $answer = $this->generate($prompt);
 
-        $toolsToCall = $this->getToolsToCall($answer);
+        $answer = $this->generate($prompt);
+        $this->handleTools($answer);
 
-        foreach ($toolsToCall as $toolToCall) {
-            $this->lastFunctionCalled = $toolToCall;
-        }
```

Comment:
I don't think this is necessary since the `lastFunctionCalled` is set by the `callFunction` method